src/main/java/com/direwolf20/buildinggadgets/client/events/EventTooltip.java

```diff
@@ -30,47 +41,37 @@
 
 @Mod.EventBusSubscriber(modid = Reference.MODID, bus = Mod.EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
 public class EventTooltip {
-
+    private static final Comparator<Multiset.Entry<IUniqueObject<?>>> ENTRY_COMPARATOR = Comparator
+            .<Multiset.Entry<IUniqueObject<?>>, Integer>comparing(Entry::getCount)
+            .reversed()
+            .thenComparing(e -> e.getElement().getObjectRegistryName());
     private static final int STACKS_PER_LINE = 8;
     private static RemoteInventoryCache cache = new RemoteInventoryCache(true);
 
-    public static void setCache(Multiset<UniqueItem> cache) {
+    public static void setCache(Multiset<com.direwolf20.buildinggadgets.common.util.tools.UniqueItem> cache) {
         EventTooltip.cache.setCache(cache);
     }
 
-    private static void tooltipIfShift(@SuppressWarnings("unused") List<ITextComponent> tooltip, Runnable r) {
-        if (Screen.hasShiftDown())
-            r.run();
-    }
-
     public static void addTemplatePadding(ItemStack stack, List<ITextComponent> tooltip) {
         //This method extends the tooltip box size to fit the item's we will render in onDrawTooltip
         Minecraft mc = Minecraft.getInstance();
-        if (stack.getItem() instanceof ITemplate) {
-            ITemplate template = (ITemplate) stack.getItem();
-            String UUID = template.getUUID(stack);
-            if (UUID == null) return;
-
-            Multiset<UniqueItem> itemCountMap = template.getItemCountMap(stack);
-
-            Map<ItemStack, Integer> itemStackCount = new HashMap<>();
-            for (Multiset.Entry<UniqueItem> entry : itemCountMap.entrySet()) {
-                ItemStack itemStack = new ItemStack(entry.getElement().getItem(), 1);
-                itemStackCount.put(itemStack, entry.getCount());
-            }
-            List<Map.Entry<ItemStack, Integer>> list = new ArrayList<>(itemStackCount.entrySet());
-
-            int totalMissing = 0;
-            //Look through all the ItemStacks and draw each one in the specified X/Y position
-            for (Map.Entry<ItemStack, Integer> entry : list) {
-                int hasAmt = InventoryHelper.countItem(entry.getKey(), Minecraft.getInstance().player, cache);
-                if (hasAmt < entry.getValue())
-                    totalMissing = totalMissing + Math.abs(entry.getValue() - hasAmt);
-            }
-
-            int count = (totalMissing > 0) ? itemStackCount.size() + 1 : itemStackCount.size();
-            if (count > 0)
-                tooltipIfShift(tooltip, () -> {
+        if (mc.world == null) //populateSearchTreeManager...
+            return;
+        mc.world.getCapability(CapabilityTemplate.TEMPLATE_PROVIDER_CAPABILITY).ifPresent(provider -> {
+            stack.getCapability(CapabilityTemplate.TEMPLATE_KEY_CAPABILITY).ifPresent(templateKey -> {
```

Comment:
This is not 2 Systems... It's 2 interfaces which form the provider System. 
Here the explanation again for protocol:
- Templates are tol large to be stored on the Item directly, therefore they need to be stored in the world save
- Therefore we are storing only a uuid on the item and query the Template from the provider
    - this cannot be simplified into a single Template cap, because in initCapabilities its impossible to decide wheter you are on th Server
- As by suggestion of gigaherz and diesieben07 I made the save+client cache a world capability and referenced it from the Item
    - whilst I technically no longer need the key (it used to be resposible for creating empty templates), I find it handy to do an ifPresent and have it ready instead of being required to call some static method on the Item
    - (I think this is more elegant...)

Reply:
Sorry, this was left before we spoke